packages/catalog-client/src/types/api.ts

```diff
@@ -269,6 +280,68 @@ export type AddLocationResponse = {
   exists?: boolean;
 };
 
+/**
+ * The request type for {@link CatalogClient.getPaginatedEntities}.
+ *
+ * @alpha
+ */
+export type GetPaginatedEntitiesRequest =
+  | GetPaginatedEntitiesInitialRequest
+  | GetPaginatedEntitiesCursorRequest;
+
+/**
+ * A request type for {@link CatalogClient.getPaginatedEntities}.
+ * The method takes this type in an initial pagination request,
+ * when requesting the first batch of entities.
+ *
+ * The properties filter, sortField, query and sortFieldOrder, are going
+ * to be immutable for the entire lifecycle of the following requests.
+ *
+ * @alpha
+ */
+export type GetPaginatedEntitiesInitialRequest =
+  | {
+      fields?: string[];
+      limit?: number;
+      filter?: EntitiesFilter;
+      sortFields?: EntitySortField[];
+      query?: string;
+    }
+  | undefined;
```

Comment:
Having undefined as part of the type definition reduces it's reusability.

Can we remove allowing this to be undefined?

Reply:
the idea was to treat `undefined` as an initial request. But I like it, good idea.
Fixed in https://github.com/backstage/backstage/pull/12246/commits/2f2baf4b6513d671425296994c99bc5a9d790171